dashboard/data/file_preprocessing/echo_files_parser.py

```diff
@@ -70,25 +69,43 @@ def parse_files(self, echo_files: tuple[str, io.StringIO]) -> EchoFilesParser:
             exception_dfs.append(exceptions_df)
             echo_dfs.append(echo_df)
 
-        if echo_df is not None:
+        if echo_dfs:
             self.echo_df = pd.concat(echo_dfs, ignore_index=True)
-        self.exceptions_df = pd.concat(exception_dfs, ignore_index=True)
+        if exception_dfs:
+            self.exceptions_df = pd.concat(exception_dfs, ignore_index=True)
 
         return self
 
+    def merge_eos(self, eos_df: pd.DataFrame) -> None:
+        """
+        Merge echo df with eos df by plate and well
+
+        :param eos_df: dataframe with eos, plate and well
+        """
+        # handle different well naming (A01 or A1)
+        eos_df["Well"] = eos_df["Well"].str.replace(r"0(?!$)", "", regex=True)
+        self.echo_df["Source Well"] = self.echo_df["Source Well"].str.replace(
+            r"0(?!$)", "", regex=True
+        )
+
+        left_cols = ["Source Plate Barcode", "Source Well"]
+        right_cols = ["Plate", "Well"]
+        merged_df = pd.merge(
+            self.echo_df, eos_df, how="left", left_on=left_cols, right_on=right_cols
+        )
```

Comment:
On the meeting Krzysztof mentioned we could discard compounds w/o eos number, as ideally they're all should be included. So just an info about missing ids will suffice (and then we could do an inner join here if I understand it correctly)

Reply:
Done, didn't really have idea where to place the alert...
![image](https://github.com/zuzg/drug-screening/assets/81712088/296b26b0-07b0-4a55-a493-c982748a2b8b)